packages/tools/src/utilities/stackPrefetch/stackContextPrefetch.ts

```diff
@@ -0,0 +1,386 @@
+import {
+  getEnabledElement,
+  StackViewport,
+  imageLoader,
+  Enums,
+  eventTarget,
+  imageLoadPoolManager,
+  cache,
+  getConfiguration as getCoreConfiguration,
+} from '@cornerstonejs/core';
+import { addToolState, getToolState } from './state';
+
+const requestType = Enums.RequestType.Prefetch;
+const priority = 0;
+
+let configuration = {
+  maxImagesToPrefetch: Infinity,
+  // Fetch up to 1 image before and 50 after
+  minBefore: 1,
+  maxAfter: 5,
+  preserveExistingPool: false,
+};
+
+let resetPrefetchTimeout;
+// Starting the prefetch quickly isn't an issue as the main image is already being
+// loaded, so a 5 ms prefetch delay is fine
+const resetPrefetchDelay = 5;
+
+function range(lowEnd, highEnd) {
+  // Javascript version of Python's range function
+  // http://stackoverflow.com/questions/3895478/does-javascript-have-a-method-like-range-to-generate-an-array-based-on-suppl
+  lowEnd = Math.round(lowEnd) || 0;
+  highEnd = Math.round(highEnd) || 0;
+
+  const arr = [];
+  let c = highEnd - lowEnd + 1;
+
+  if (c <= 0) {
+    return arr;
+  }
+
+  while (c--) {
+    arr[c] = highEnd--;
+  }
+
+  return arr;
+}
+
+function getStackData(element) {
+  const enabledElement = getEnabledElement(element);
+
+  if (!enabledElement) {
+    throw new Error(
+      'stackPrefetch: element must be a valid Cornerstone enabled element'
+    );
+  }
+
+  const { viewport } = enabledElement;
+
+  if (!(viewport instanceof StackViewport)) {
+    throw new Error(
+      'stackPrefetch: element must be a StackViewport, VolumeViewport stackPrefetch not yet implemented'
+    );
+  }
+
+  return {
+    currentImageIdIndex: viewport.getCurrentImageIdIndex(),
+    imageIds: viewport.getImageIds(),
+  };
+}
+
```

Comment:
this is shared between two prefetchers, can we refactor in to stackprefetch/utils?